src/oci.rs

```diff
@@ -0,0 +1,247 @@
+use std::ffi::CString;
+
+extern crate libloading as lib;
+
+fn lib_file() -> String {
+    #[cfg(target_os = "linux")]
+    {
+        String::from("shared/libwasm2oci.so")
+    }
+    #[cfg(target_os = "macos")]
+    {
+        String::from("shared/libwasm2oci.dylib")
+    }
+    #[cfg(target_os = "windows")]
+    {
+        String::from("shared/libwasm2oci.dll")
+    }
+}
+
+pub fn pull_wasm(reference: String, file: String) -> Result<(), OCIError> {
+    println!("pulling {:?} into {:?}", reference, file);
+    let c_ref = CString::new(reference)?;
+    let c_file = CString::new(file)?;
+
+    let go_str_ref = GoString {
+        p: c_ref.as_ptr(),
+        n: c_ref.as_bytes().len() as isize,
+    };
+    let go_str_file = GoString {
+        p: c_file.as_ptr(),
+        n: c_file.as_bytes().len() as isize,
+    };
+
+    let lib = lib::Library::new(lib_file())?;
+    unsafe {
+        let func: lib::Symbol<unsafe extern "C" fn(r: GoString, f: GoString) -> i64> =
+            lib.get(b"Pull")?;
+        match func(go_str_ref, go_str_file) {
+            0 => return Ok(()),
+            _ => return Err(OCIError::Custom(String::from("cannot pull module"))),
+        }
+    }
+}
+
+pub fn push_wasm(reference: &str, file: &str) -> lib::Result<i64> {
```

Comment:
Thinking about this a bit more, we probably don't need a push function for the node.